sklearn/preprocessing/data.py

```diff
@@ -2210,6 +2213,17 @@ def __init__(self, n_quantiles=1000, output_distribution='uniform',
         self.random_state = random_state
         self.copy = copy
 
+    @staticmethod
+    def _nanpercentile_force_finite(column_data, percentiles):
+        """Force the output of nanpercentile to be finite."""
+        percentile = nanpercentile(column_data, percentiles)
+        with np.errstate(invalid='ignore'):  # hide NaN comparison warnings
+            if np.all(np.isclose(percentile, np.nan, equal_nan=True)):
+                warnings.warn("All samples in a column of X are NaN.")
```

Comment:
Maybe you can mention in the warning that you are returning 0 for all the quantiles?

Reply:
I am not sure anymore why I force percentile to be finite.